File: backend/seed_db.py
import os
import sys
from dotenv import load_dotenv
from pymongo import MongoClient
from datetime import datetime
from seed_data import mock_profiles

# Load environment variables
load_dotenv()

def seed_database():
    try:
        # MongoDB Atlas connection
        mongo_uri = os.getenv("MONGODB_URI")
        if not mongo_uri:
            print("Error: MONGODB_URI environment variable not set")
            sys.exit(1)
            
        client = MongoClient(mongo_uri)
        db = client[os.getenv("MONGODB_DB_NAME", "sportmatch")]
        
        # Check if profiles collection is empty
        if db.profiles.count_documents({}) == 0:
            print("Seeding profiles collection...")
            profiles = []
            for profile in mock_profiles:
                profile["createdAt"] = datetime.now()
                profile["updatedAt"] = datetime.now()
                profiles.append(profile)
                
            if profiles:
                result = db.profiles.insert_many(profiles)
                print(f"Successfully inserted {len(result.inserted_ids)} profiles")
            else:
                print("No profiles to insert")
        else:
            print("Profiles collection is not empty. Skipping seed.")
            
        print("Database seeding completed successfully!")
            
    except Exception as e:
        print(f"Error seeding database: {str(e)}")
        sys.exit(1)

if __name__ == "__main__":
    seed_database()

File: scripts/seed-db.ts
import { MongoClient } from "mongodb"
import { mockProfiles } from "@/data/mock-profiles"

// This script seeds the MongoDB database with initial data
async function seedDatabase() {
  if (!process.env.MONGODB_URI) {
    throw new Error("Please add your MongoDB URI to .env.local")
  }

  const client = new MongoClient(process.env.MONGODB_URI)

  try {
    await client.connect()
    console.log("Connected to MongoDB")

    const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

    // Check if profiles collection already has data
    const profilesCount = await db.collection("profiles").countDocuments()

    if (profilesCount === 0) {
      console.log("Seeding profiles collection...")

      // Transform mock profiles to MongoDB format
      const profiles = mockProfiles.map((profile) => ({
        name: profile.name,
        age: profile.age,
        location: profile.location,
        bio: profile.bio,
        sports: profile.sports,
        distance: profile.distance,
        profilePicture: profile.profilePicture,
        createdAt: new Date(),
        updatedAt: new Date(),
      }))

      await db.collection("profiles").insertMany(profiles)
      console.log(`Inserted ${profiles.length} profiles`)
    } else {
      console.log(`Profiles collection already has ${profilesCount} documents. Skipping seed.`)
    }

    console.log("Database seeding completed")
  } catch (error) {
    console.error("Error seeding database:", error)
  } finally {
    await client.close()
  }
}

// Run the seed function
seedDatabase()
  .then(() => process.exit(0))
  .catch((err) => {
    console.error(err)
    process.exit(1)
  })
